tests/test_core/test_error_handlers.py

```
@@ -0,0 +1,243 @@
+
+    @app.route("/test/rate-limit-error")
+    def raise_rate_limit_error():
+        raise RateLimitError("Rate limit exceeded", retry_after=60)
+
+    @app.route("/test/external-api-error")
+    def raise_external_api_error():
+        raise ExternalAPIError("test_service", "External API failed")
+
+    @app.route("/test/llm-error")
+    def raise_llm_error():
+        raise LLMError("LLM failed", provider="test_provider")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)